### PR TITLE
Add options to ssh connection

### PR DESCRIPTION
**Pull request by mpekatsoula**
Signed-off-by: Michalis Spyrou <michalis.spyrou@arm.com>

### CODE REVIEW

File: devlib/utils/ssh.py [hunk 1]
```diff
@@ -247,7 +249,8 @@ def background(self, command, stdout=subprocess.PIPE, stderr=subprocess.PIPE, as
             keyfile_string = '-i {}'.format(self.keyfile) if self.keyfile else ''
             if as_root and not self.connected_as_root:
                 command = self.sudo_cmd.format(command)
-            command = '{} {} {} {}@{} {}'.format(ssh, keyfile_string, port_string, self.username, self.host, command)
+            options = " ".join([ "-o {}={}".format(key,val) for key,val in self.options.items()])
+            command = '{} {} {} {} {}@{} {}'.format(ssh, options, keyfile_string, port_string, self.username, self.host, command)
```
Comment (marcbonnici) [inline devlib/utils/ssh.py]: Can this be line wrapped?

Comment (mpekatsoula) [reply devlib/utils/ssh.py]: Done

File: devlib/utils/ssh.py [hunk 2]
```diff
@@ -247,7 +249,9 @@ def background(self, command, stdout=subprocess.PIPE, stderr=subprocess.PIPE, as
             keyfile_string = '-i {}'.format(self.keyfile) if self.keyfile else ''
             if as_root and not self.connected_as_root:
                 command = self.sudo_cmd.format(command)
-            command = '{} {} {} {}@{} {}'.format(ssh, keyfile_string, port_string, self.username, self.host, command)
+            options = " ".join([ "-o {}={}".format(key,val) for key,val in self.options.items()])
+            command = '{} {} {} {} {}@{} {}'.format(ssh, options, keyfile_string, port_string, \
```
Comment (setrofim) [inline devlib/utils/ssh.py]: Do not need `\` inside parens. Also could you please also break up the other lines that are too wide to fit in the GitHub review viewport without scrolling.

File: doc/connection.rst [hunk 3]
```diff
@@ -119,7 +119,7 @@ Connection Types
 
 .. class:: SshConnection(host, username, password=None, keyfile=None, port=None,\
                          timeout=None, password_prompt=None, \
-                         sudo_cmd="sudo -- sh -c {}")
+                         sudo_cmd="sudo -- sh -c {}, options=None")
```
Comment (setrofim) [inline doc/connection.rst]: Shouldn't the `, options=None` be ouside the `"`?

Comment (mpekatsoula) [reply doc/connection.rst]: Oops, yes.

**Pull request merged**